src/lib/rpcHandler.js

```diff
@@ -0,0 +1,136 @@
+'use strict'
+const BaseError = require('five-bells-shared/errors/base-error')
+
+class RpcHandler {
+  constructor (params, websocket, requestingUser) {
+    this.log = params.log
+    this.uri = params.uriManager
+    this.validator = params.validator
+    this.notificationBroadcaster = params.notificationBroadcaster
+
+    this.websocket = websocket
+    this.requestingUser = requestingUser
+    this.accountSubscriptions = []
+    this.sendNotification = this._sendNotification.bind(this)
+
+    websocket.on('message', this.handleMessage.bind(this))
+    websocket.on('close', this.removeAccountSubscriptions.bind(this))
+    this._send({ jsonrpc: '2.0', id: null, method: 'connect' })
+  }
+
+  handleMessage (reqMessageString) {
+    const resMessage = { jsonrpc: '2.0', id: null }
+    try {
+      const reqMessage = JSON.parse(reqMessageString)
+      const validatorResult = this.validator.create('RpcRequest')(reqMessage)
+      if (!validatorResult.valid) {
+        throw new RpcError(-32600, 'Invalid Request', {validationErrors: validatorResult.errors})
+      }
+      if (reqMessage.id === null) throw new RpcError(40000, 'Invalid id')
+      resMessage.id = reqMessage.id
+
+      if (reqMessage.method === 'subscribe_account') {
+        resMessage.result = this.subscribeAccount(reqMessage.params.eventType,
+          reqMessage.params.accounts)
+      } else {
+        throw new RpcError(-32601, 'Unknown method: ' + reqMessage.method)
+      }
+    } catch (err) {
+      resMessage.error = {
+        code: err instanceof SyntaxError ? -32700 : (err.code || 50000),
```

Comment:
Instead of having these codes scattered throughout the code it would be better to have one file that's required in a number of places that would export constants like `websocketErrorCodes.SYNTAX_ERROR`